src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunction.java

```diff
@@ -507,36 +482,47 @@ public static ApartmentValueFunction getRandomApartmentValueFunction() {
     int pricePerNightStartBound = random.nextInt(pricePerNightEndBound);
     int nbMinNightStartBound = random.nextInt(nbMinNightEndBound);
 
-    apartValueFunction.setFloorAreaValueFunction(
-        new LinearValueFunction(floorAreaStartBound, floorAreaEndBound));
-    apartValueFunction.setNbBedroomsValueFunction(nbBedroomsEndBoundMap);
-    apartValueFunction.setNbSleepingValueFunction(nbSleepingEndBoundMap);
-    apartValueFunction.setNbBathroomsValueFunction(nbBathroomsEndBoundMap);
-    apartValueFunction.setTerraceValueFunction(new BooleanValueFunction(terraceEndBound));
-    apartValueFunction.setFloorAreaTerraceValueFunction(
+    apartValueFunction.setInternalDoubleValueFunction(
+        Criterion.FLOOR_AREA, new LinearValueFunction(floorAreaStartBound, floorAreaEndBound));
+    apartValueFunction.setInternalDoubleValueFunction(Criterion.NB_BEDROOMS, nbBedroomsEndBoundMap);
+    apartValueFunction.setInternalDoubleValueFunction(Criterion.NB_SLEEPING, nbSleepingEndBoundMap);
+    apartValueFunction.setInternalDoubleValueFunction(
+        Criterion.NB_BATHROOMS, nbBathroomsEndBoundMap);
+    apartValueFunction.setInternalBooleanValueFunction(
+        Criterion.TERRACE, new BooleanValueFunction(terraceEndBound));
+    apartValueFunction.setInternalDoubleValueFunction(
+        Criterion.FLOOR_AREA_TERRACE,
         new LinearValueFunction(floorAreaTerraceStartBound, floorAreaTerraceEndBound));
-    apartValueFunction.setWifiValueFunction(new BooleanValueFunction(wifiEndBound));
-    apartValueFunction.setPricePerNightValueFunction(
+    apartValueFunction.setInternalBooleanValueFunction(
+        Criterion.WIFI, new BooleanValueFunction(wifiEndBound));
+    apartValueFunction.setInternalDoubleValueFunction(
+        Criterion.PRICE_PER_NIGHT,
         new LinearValueFunction(pricePerNightStartBound, pricePerNightEndBound));
-    apartValueFunction.setNbMinNightValueFunction(
+    apartValueFunction.setInternalDoubleValueFunction(
+        Criterion.NB_MIN_NIGHT,
         new ReversedLinearValueFunction(nbMinNightStartBound, nbMinNightEndBound));
-    apartValueFunction.setTeleValueFunction(new BooleanValueFunction(teleEndBound));
+    apartValueFunction.setInternalBooleanValueFunction(
+        Criterion.TELE, new BooleanValueFunction(teleEndBound));
 
     List<Double> weightRange = RandomRange.weightRangeOfSum(1d, 10);
```

Comment:
Remplacer 10 par le nombre de critères pour expliquer d’où vient ce nombre. OPT.

Reply:
fix in f338d5ca96f1cc2ad77929e19a335c8c2f905771 
```java
Criterion.values().length
```